tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+## RTE_Components.h
+
+The file `./RTE/RTE_Components.h` is automatically created by the CMSIS Project Manager (during CONVERT). For each selected Software Component it contains `#define` statements required by the component. These statements are defined in the \*.PDSC file for that component. The following example shows a sample content of a RTE_Components.h file:
+
+```c
+/* Auto generated Run-Time-Environment Component Configuration File *** Do not modify ! *** */
+
+#ifndef RTE_COMPONENTS_H
+#define RTE_COMPONENTS_H
+
+/* Define the Device Header File: */
+#define CMSIS_device_header "stm32f10x.h"
+
+#define RTE_Network_Interface_ETH_0     /* Network Interface ETH 0 */
+#define RTE_Network_Socket_BSD          /* Network Socket BSD */
+#define RTE_Network_Socket_TCP          /* Network Socket TCP */
+#define RTE_Network_Socket_UDP          /* Network Socket UDP */
+
+#endif /* RTE_COMPONENTS_H */
+```
+
+The typical usage of the `RTE_Components.h` file is in header files to control the inclusion of files that are related to other components of the same Software Pack.
+```c
+#include "RTE_Components.h"
+#include  CMSIS_device_header
+
+#ifdef  RTE_Network_Interface_ETH_0  // if component Network Interface ETH 0 is included
+#include "Net_Config_ETH_0.h"        // add the related configuration file for this component
+#endif
+```
+
+
+
+# YML Format
+
+## Name Conventions
+
+### Component Name Conventions
+
+The CMSIS Project Manager uses the following syntax to specify component names:
+
+```text
+[Cvendor::] Cclass [&Cbundle] :Cgroup [:Csub] [&Cvariant] [@[>=]Cversion]
+```
+
+Element    | Description
+:----------|:---------------------
+`Cvendor`  | is the name of the component vendor defined in `<components>` element of the software pack (optional).
+`Cclass`   | is the component class name  defined in `<components>` element of the software pack (required)
+`Cbundle`  | is the bundle name of component class defined in `<bundle>` element of the software pack (optional).
+`Cgroup`   | is the component group name  defined in `<components>` element of the software pack (required).
+`Csub`     | is the component sub-group name  defined in `<components>` element of the software pack (optional).
+`Cvariant` | is the component sub-group name  defined in `<components>` element of the software pack (optional).
+`Cversion` | is the version number of the component, with `@1.2.3` that must exactly match, or `@>=1.2.3` that allows any version higher or equal.
+
+**Notes:**
+
+- The unique separator `::` allows it to omit `Cvendor`
+- When `Cvariant` is omitted, the default `Cvariant` is selected.
+
+**Examples:**
+
+```yml
+component: ARM::CMSIS:CORE                           # CMSIS Core component from vendor ARM (any version)
+component: ARM::CMSIS:CORE@5.5.0                     # CMSIS Core component from vendor ARM (with version 5.5.0)
```

Comment:
how does it work with the conditions ?

A condition in the pdsc may very well impose a version typically.

Reply:
When a condition creates a conflict with an explicit requested version, the ProjMgr should generate an error.